File: model/c#/RealmModels.cs
using System;
using System.Collections.Generic;
using Realms;

public class Account : RealmObject
{
    [Required]
    public string provider { get; set; }

    [Required]
    public string provider_id { get; set; }

    public string data { get; set; }

    public IList<Token> tokens { get; }

    public User user { get; set; }
}

public class ClusterNode : RealmObject
{
    [PrimaryKey]
    [Required]
    public string id { get; set; }

    public IList<RealmFile> files { get; }
}

public class Metadata : RealmObject
{
    [Required]
    public string userId { get; set; }

    [Required]
    public string namespace { get; set; }

    [Required]
    public string key { get; set; }

    [Required]
    public string value { get; set; }
}

public class Permission : RealmObject
{
    public User user { get; set; }

    public RealmFile file { get; set; }

    public bool mayRead { get; set; }

    public bool mayWrite { get; set; }

    public bool mayManage { get; set; }
}

public class RealmFile : RealmObject
{
    [PrimaryKey]
    [Required]
    public string id { get; set; }

    [Required]
    public string path { get; set; }

    public User owner { get; set; }

    public ClusterNode cluster_node { get; set; }
}

public class Token : RealmObject
{
    [PrimaryKey]
    [Required]
    public string token { get; set; }

    public DateTimeOffset expires { get; set; }

    public DateTimeOffset? revoked { get; set; }

    public IList<RealmFile> files { get; }

    public Account account { get; set; }

    [Required]
    public string app_id { get; set; }
}

public class User : RealmObject
{
    [PrimaryKey]
    [Required]
    public string id { get; set; }

    public IList<Account> accounts { get; }

    public bool isAdmin { get; set; }
}

File: model/swift/RealmModels.swift
import Foundation
import RealmSwift

class Account: Object {
  dynamic var provider = ""
  dynamic var provider_id = ""
  dynamic var data: String?
  let tokens = List<Token>()
  dynamic var user: User?

  override static func indexedProperties() -> [String] {
    return [
      "provider_id",
    ]
  }
}

class ClusterNode: Object {
  dynamic var id = ""
  let files = List<RealmFile>()

  override static func primaryKey() -> String? {
    return "id"
  }
}

class Metadata: Object {
  dynamic var userId = ""
  dynamic var namespace = ""
  dynamic var key = ""
  dynamic var value = ""

  override static func indexedProperties() -> [String] {
    return [
      "userId",
    ]
  }
}

class Permission: Object {
  dynamic var user: User?
  dynamic var file: RealmFile?
  dynamic var mayRead = false
  dynamic var mayWrite = false
  dynamic var mayManage = false
}

class RealmFile: Object {
  dynamic var id = ""
  dynamic var path = ""
  dynamic var owner: User?
  dynamic var cluster_node: ClusterNode?

  override static func primaryKey() -> String? {
    return "id"
  }
}

class Token: Object {
  dynamic var token = ""
  dynamic var expires = NSDate()
  dynamic var revoked: NSDate?
  let files = List<RealmFile>()
  dynamic var account: Account?
  dynamic var app_id = ""

  override static func primaryKey() -> String? {
    return "token"
  }
}

class User: Object {
  dynamic var id = ""
  let accounts = List<Account>()
  dynamic var isAdmin = false

  override static func primaryKey() -> String? {
    return "id"
  }
}

